### ISSUE TITLE
array of objects completion: line return indentation should be always at array item level

### ISSUE THREAD

**Issue by fbaligand** [redhat-developer/vscode-yaml#430]
### Is your enhancement related to a problem? Please describe.
Currently, when I have an array of objects, if I type RETURN on first item of my array, new line indentation is at parent level (not at array item level).

For example, if I type:
``` yaml
- prop1: value
```
And then ENTER, I get this:
``` yaml
- prop1: value
#INDENTATION AT ROOT
```

Although if I have this:
For example, if I type:
``` yaml
- prop1: value
```
And then ENTER, I get this:
``` yaml
- prop1: value
  prop2: value
```
And then ENTER, I get this:
``` yaml
- prop1: value
  prop2: value
  #INDENTATION AT ARRAY ITEM LEVEL
```
  

### Describe the solution you would like
I would like that always, when array is an "array of objects", RETURN generates a new line with "array item indentation"

**Comment by evidolob** [comment]
@fbaligand You can achieve that if you enable `editor.formatOnType` for yaml files. We have `formatOnType` support starting with `0.15.0` version.

**Comment by fbaligand** [comment]
Thanks for the tip. 

That said, this would be great if editor detects that current line is an array property (`- key: value`), and so, generates new line at array item level.

Having a global configuration for that is nice, but in a same yaml, there could have both an array of simple values and an array of objects.
And a global configuration can’t deal with both cases.

**Comment by evidolob** [comment]
Have you try to enable `formatOnType`? I mean that if it works not like you expect to work it is a bug.
for this `- key: value\n` it will generate:
```yaml
- key: value
  |-> cursor
```

**Comment by fbaligand** [comment]
I just tried to put this in VSCODE settings.json:
`"editor.formatOnType": true`

But the behavior is still the same.

**Comment by evidolob** [comment]
OK, now I understand what you propose to do.
To achieve that we need to analyze JSON Schema to see what type of elements it expect to be present in array.

**Comment by fbaligand** [comment]
Yes, that would be great: detect that array item type is “object”.
